app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -73,49 +74,71 @@ def batch_callbacks(batch)
     end
   end
 
-  def delete_associations
-    destroyable_associations.each do |association|
-      ReflectionDestroyer.new(object, association, caller_worker_hierarchy).destroy_later
+  def destroy_and_delete_associations
+    Array(object.background_deletion).each do |association, options|
+      ReflectionDestroyer.new(object, association, options, caller_worker_hierarchy).destroy_later if destroyable_association?(association)
     end
   end
 
-  private
-
-  def destroy_method
-    @destroy_method.presence || 'destroy'
+  def destroyable_association?(_association)
+    true
   end
 
+  private
+
   def called_from_provider_hierarchy?
     return unless (tenant_id = object.tenant_id)
     caller_worker_hierarchy.include?("Hierarchy-Account-#{tenant_id}")
   end
 
-  def destroyable_associations
-    object.class.reflect_on_all_associations.select { |reflection| destroyable_association?(reflection) }
-  end
-
-  def destroyable_association?(reflection)
-    %i[destroy delete_all].include?(reflection.options[:dependent])
-  end
-
   def callback_options
     { 'object_global_id' => object.to_global_id, 'caller_worker_hierarchy' => caller_worker_hierarchy }
   end
 
   class ReflectionDestroyer
 
-    def initialize(main_object, reflection, caller_worker_hierarchy)
+    class FakeReflection
```

Comment:
'fake' 😂 😂 😂 I think it is better for example 'SimpleReflection', among many other options

Reply:
hehe :D could be..